packages/neon/neon/lib/src/settings/models/select_option.dart

```diff
@@ -46,23 +44,27 @@ class SelectOption<T> extends Option<T> {
       return stored as T;
     }
 
-    return _fromString(vs, stored);
+    return _deserialize(vs, stored);
   }
 
-  static T? _fromString<T>(final Map<T, LabelBuilder> vs, final String? valueStr) {
-    if (valueStr == null) {
-      return null;
-    }
+  @override
+  void reset() {
+    // ignore: discarded_futures
+    storage.remove(key);
 
-    return vs.keys.firstWhereOrNull((final e) => e.toString() == valueStr);
+    super.reset();
   }
 
   Map<T, LabelBuilder> _values;
 
   @override
   set value(final T value) {
     super.value = value;
-    unawaited(storage.setString(key, serialize()));
+
+    if (value != null) {
+      // ignore: discarded_futures
+      storage.setString(key, serialize()!);
```

Comment:
Better use unawaited

Reply:
Not possible as the type is FutureOr and unawaited needs a plain future.

I quite like using this in interfaces but I should just revert that :thinking:

Reply:
Not possible as the type is FutureOr and unawaited needs a plain future.

I quite like using this in interfaces but I should just revert that :thinking: